warrior/Framework/Utils/cli_Utils.py

```diff
@@ -174,125 +113,88 @@ def connect_telnet(ip, port="23", username="", password="",
-                      "is not found in the system response: {2}"\
-                      .format(command, prompt, child.before), "error")
-                break
-    except Exception as exception:
-        print_error(" ! could not connect to %s...check logs" % ip)
-        print_exception(exception)
-    else:
-        return telnetobj, conn_string
+                  "in 'warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py'")
+
+    wc_obj = WNetwork.warrior_cli_class.WarriorCli()
+    session_object, conn_string = wc_obj.connect_telnet(
+        ip=ip, port=port, username=username, password=password, logfile=logfile, timeout=timeout,
+        prompt=prompt, conn_options=conn_options, custom_keystroke=custom_keystroke, escape=escape)
+
+    return session_object, conn_string
 
 
 def disconnect_telnet(child):
     """Disconnects a telnet session """
     print_warning("This method is obsolete and will be deprecated soon. Please"
                   " use 'disconnect_telnet' method of 'PexpectConnect' class "
-                  "in 'warrior/Framework/ClassUtils/warrior_connect_class.py'")
-
-    time.sleep(2)
-    child.sendcontrol(']')
-    time.sleep(2)
-    child.expect('telnet> ')
-    time.sleep(2)
-    child.sendline('q')
-    time.sleep(2)
-    child.close()
+                  "in 'warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py'")
+
+    if isinstance(child, WNetwork.warrior_cli_class.WarriorCli):
+        child = child.disconnect_telnet()
+    elif isinstance(child, pexpect.spawn):
+        wc_obj = WNetwork.warrior_cli_class.WarriorCli()
+        wc_obj.conn_obj = WNetwork.warrior_cli_class.PexpectConnect()
+        wc_obj.conn_obj.target_host = child
+        wc_obj.disconnect_telnet()
+
     return child
 
 
 def disconnect(child):
     """
-    - Disconnects a pexpect session
+    - Disconnects warrior_cli_class session object(pexpect/paramiko)
     - Returns session object(same child)
     """
     print_warning("This method is obsolete and will be deprecated soon. Please"
                   " use 'disconnect' method of 'PexpectConnect' class "
-                  "in 'warrior/Framework/ClassUtils/warrior_connect_class.py'")
+                  "in 'warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py'")
+
+    if isinstance(child, WNetwork.warrior_cli_class.WarriorCli):
+        child = child.disconnect()
+    elif isinstance(child, pexpect.spawn):
+        wc_obj = WNetwork.warrior_cli_class.WarriorCli()
+        wc_obj.conn_obj = WNetwork.warrior_cli_class.PexpectConnect()
+        wc_obj.conn_obj.target_host = child
+        wc_obj.disconnect()
 
-    if child.isalive():
-        if child.ignore_sighup:
-            child.ignore_sighup = False
-        child.close()
     return child
 
-@cmdprinter
-def send_command_and_get_response(sessionobj, prompt1, prompt2, command):
+
+def send_command_and_get_response(session_object, start_prompt, end_prompt, command):
```

Comment:
do not change names of existing arguments.. because users may be using it as named arguments in their function calls like send_command_and_get_response(session_object, prompt1='.*', prompt2='\$', command=cmd) and changing name will break this function call.